enhancements/builds/multi-arch-builds.md

```diff
@@ -0,0 +1,456 @@
+reduce "noise" in the list of tags reported by registries which support listing
+the tags in a repository.
+
+### Workflow Description
+
+#### Variation 1/3: SCM integrated
+1. Application developer merges a PR to their repository's main branch.
+1. The SCM platform notifies OpenShift via a webhook.
+1. OpenShift has previously associated the webhook with triggering a BuildConfig.
+
+#### Variation 2/3: not SCM integrated
+1. Application developer manually triggers a BuildConfig.
+
+#### Variation 3/3: not SCM integrated
+1. Application developer creates a Build.
+
+#### Common
+1. The openshift-controller-manager starts N+1 build pods.
+1. Platform-specific images are pushed to the registry.
+1. The build completes, and a manifest list is pushed to the registry.
+1. Platform-specific images are untagged from the registry.
+
+#### Variation 1/3: no registry/DeploymentConfig integration
+1. Application developer retrieves a pullable reference to the manifest list from the build status object and updates their pod specs.
+1. Pods are restarted.
+
+#### Variation 2/3: registry/DeploymentConfig integration
+1. The registry notifies OpenShift of the newly pushed tag.
+1. DeploymentConfigs which reference this tag in an ImageChangeTrigger cause the image to be re-pulled and pods to be restarted on affected nodes.
+
+#### Variation 3/3: we're just here to build
+1. Application developer retrieves a pullable reference to the manifest list from the build's status fields and pulls some subset of images from the registry from elsewhere.
+
+```mermaid
+flowchart
+A(Developer merges changes to SCM) --> B(SCM notifies OpenShift webhook)
+B --> C(OpenShift webhook triggers OpenShift BuildConfig)
+C --> E
+D(Developer triggers OpenShift BuildConfig) --> E
+R(Developer creates OpenShift Build) --> E
+
+E(openshift-controller-manager starts N+1 build pods) --> F(Build pod pushes image to registry)
+E --> N(Build pod pushes image to registry)
+N --> G
+E --> P(Build pod pushes image to registry)
+P --> G
+F --> G(Build pod pushes manifest list to registry)
+G --> Q(Build pod tags manifest list in registry)
+Q --> H(Build pod untags images in registry)
+
+Q --> I(Developer updates pod specs)
+I --> J(OpenShift restarts pods)
+
+Q --> K(Registry notifies OpenShift webhook) --> L(OpenShift triggers DeploymentConfig)
+L --> J
+
+Q --> M(Developer pulls images)
+```
+
+### API Extensions
+
+Build and BuildConfig objects (build.openshift.io/v1) will be extended to
+include a target platform list. If set, the build controller will attempt the
+build for the specified platforms. If the platform list is not set in the Build
+or BuildConfig, the build controller should check for a list in its
+OpenShiftControllerManagerConfig, encoded in the `config` configMap in the
+`openshift-controller-manager` namespace. If that is also unset, the build
+controller will attempt to build for all of the architectures it detects in the
```

Comment:
> i.e. If one is configured to be generated then it is always generated, even if only for one architecture?

that would be my expectation, yes.  perhaps it could even be renamed the "manifestlist" bool, and then when it's enabled, you must also tell us which arches you want built (or get certain cluster defaults).  You're welcome to only specify a single arch, we'll still push a manifestlist containing just that arch.

that's somewhat analogous to how imagestreams were updated to support manifestlists.